apps/ejabberd/src/mod_admin_extra_roster.erl

```diff
@@ -163,12 +163,22 @@ commands() ->
                      Group :: binary() | string(),
                      Subs :: subs()) -> 'error' | 'ok'.
 add_rosteritem(LocalUser, LocalServer, User, Server, Nick, Group, Subs) ->
-    case subscribe(LocalUser, LocalServer, User, Server, Nick, Group, Subs, []) of
-        {atomic, ok} ->
-            push_roster_item(LocalUser, LocalServer, User, Server, {add, Nick, Subs, Group}),
-            ok;
-        _ ->
-            error
+    case ejabberd_auth:is_user_exists(LocalUser, LocalServer) of
```

Comment:
This part definitely requires some refactoring as there is to many nested case clauses.